src/ServiceControl/ExternalIntegrations/EventDispatcher.cs

```diff
@@ -14,29 +14,32 @@
     using Raven.Client;
     using ServiceBus.Management.Infrastructure.Extensions;
     using ServiceBus.Management.Infrastructure.Settings;
+    using ServiceControl.Infrastructure.DomainEvents;
 
     public class EventDispatcher : FeatureStartupTask
     {
-        private static readonly ILog Logger = LogManager.GetLogger(typeof(EventDispatcher));
-        private readonly IBus bus;
-        private readonly CriticalError criticalError;
-        private readonly IEnumerable<IEventPublisher> eventPublishers;
-        private readonly Settings settings;
-        private readonly ManualResetEventSlim signal = new ManualResetEventSlim();
-        private readonly IDocumentStore store;
-        private RepeatedFailuresOverTimeCircuitBreaker circuitBreaker;
-        private IDisposable subscription;
-        private Task task;
-        private CancellationTokenSource tokenSource;
-        private Etag latestEtag = Etag.Empty;
-
-        public EventDispatcher(IDocumentStore store, IBus bus, CriticalError criticalError, Settings settings, IEnumerable<IEventPublisher> eventPublishers)
+        static ILog Logger = LogManager.GetLogger(typeof(EventDispatcher));
+        IBus bus;
```

Comment:
Why is IBus still here?

Reply:
We use it to push the integration events out to the consumers.